relay-event-normalization/src/normalize/mod.rs

```diff
@@ -918,6 +930,48 @@ impl Default for LightNormalizationConfig<'_> {
             span_description_rules: Default::default(),
             geoip_lookup: Default::default(),
             enable_trimming: false,
+            dynamic_measurements_config: None,
+        }
+    }
+}
+
+/// Container for global and project level [`MeasurementsConfig`]. The purpose is to handle
+/// the merging logic.
+#[derive(Clone, Debug)]
+pub struct DynamicMeasurementsConfig<'a> {
+    project: Option<&'a MeasurementsConfig>,
+    global: Option<&'a MeasurementsConfig>,
+}
+
+impl<'a> DynamicMeasurementsConfig<'a> {
+    /// Constructor for [`DynamicMeasurementsConfig`].
+    pub fn new(
+        project: Option<&'a MeasurementsConfig>,
+        global: Option<&'a MeasurementsConfig>,
+    ) -> Self {
+        DynamicMeasurementsConfig { project, global }
+    }
+
+    /// Returns an iterator over the union of [`BuiltinMeasurementKey`]s from both the global and
+    /// project-level [`MeasurementsConfig`]s.
+    pub fn builtin_measurement_keys(&'a self) -> impl Iterator<Item = &'a BuiltinMeasurementKey> {
+        let from_project = self.project.iter().flat_map(|c| &c.builtin_measurements);
+        let from_global = self.global.iter().flat_map(|c| &c.builtin_measurements);
+
+        from_project.chain(from_global).unique()
+    }
+
+    /// Gets the minimum value of the max amount of custom measurements from project and global level
+    /// [`MeasurementsConfig`] if at least one is available.
+    pub fn max_custom_measurements(&'a self) -> Option<&'a usize> {
```

Comment:
@iker-barriocanal  just a nit but does it make sense to return a reference to a primitive?

Reply:
I thought about this when I was coding, and I think it doesn't matter at all. The reason I added the reference is that I didn't see any reason not to, and seemed "more correct".

Reply:
In general, it's better to return `usize` directly here since that removes an indirection and allows for further optimization. You can usually opt to return `Copy` types directly, unless they are extremely large. Particularly with `usize`, the reference is just an obstacle.